src/charm.py

```diff
@@ -41,6 +43,36 @@
 CREATED_BY_LABEL = "gateway-api-integrator.charm.juju.is/managed-by"
 
 
+@lru_cache(maxsize=1)
+def get_client(field_manager: str, namespace: str) -> Client:
+    """Initialize the lightkube client with the correct namespace and field_manager.
+
+    Args:
+        field_manager (str): field manager for server side apply when patching resources.
+        namespace (str): The k8s namespace in which resources are managed.
```

Comment:
Just out of curiosity, may I know why the type hint in docstrings? I've had feedbacks before that since we already have type hints to the arguments, we don't quite need to repeat it in the docstrings :D

Reply:
It came with the doc string generator tool that I used, since I don't the the downside of having type hints there I didn't remove them. I'll remove the type hints to be consistent with other charms.